### PR TITLE
Add Hate Crime stat vars to place overview

### PR DESCRIPTION
**Pull request by rbhoot**
Adding Hate Crime Graph to place page.

Is this the correct file and method to add graphs? Should more stat vars be added?

### CODE REVIEW

**Comment by beets** [comment]
Hi Ravi - thanks for adding these. These are a good start. Were you able to see the new charts rendered? There are some steps outlined here to help, what's missing are translation strings for the newly added titles.

https://github.com/datacommonsorg/website/blob/master/docs/developer_guide.md#adding-new-charts

To run the server, you can start two processes in different terminal windows
./run_server.sh
./run_npm.sh

Note that changes to the config files require a restart of ./run_server.sh

**Comment by rbhoot** [comment]
Hi Carolyn

For some reason I hadn't located Adding Charts section while going through the guide. Thanks for pointing it out.
I tried following the instructions after adding the translation strings for `Count_CriminalIncidents_IsHateCrime` as you can see in the `89254cd` commit. The place page looks the same for California. 

while running `./scripts/extract_messages.sh` the command fails due to [wrong path](https://github.com/datacommonsorg/website/blob/master/tools/i18n/chart_config_extractor.py#L25)
Is that step not required? The Charts do not appear even after fixing the above path, and quite a few files are updated by that command.

**Comment by rbhoot** [comment]
Added relevant translation strings for all variables. Does `configmodule.py` need to be part of the commit? Additions to `NEW_STAT_VARS` are currently part of local copy only.

After all the changes, the local deployment still looks the same. 
image

To run the server, I am using `./run_server -e lite` because of permissions issue.

File: server/config/chart_config/crime.json [hunk 1]
```diff
@@ -62,5 +62,69 @@
...
+      "scaling": 100000
+    }
+  },
+  {
+    "category": "Crime",
+    "titleId": "CHART_TITLE-Hate_Crime_Gender",
+    "title": "Hate Crime motivated by gender",
+    "description": "Total number of hate crime incidents motivated by gender",
+    "statsVars": ["Count_CriminalIncidents_BiasMotivationGender_IsHateCrime"],
+    "isOverview": false
```
Comment (beets) [inline server/config/chart_config/crime.json:88]: please add relatedCharts for this and the remaining charts you've added

File: static/js/i18n/strings/en/stats_var_labels.json [hunk 2]
```diff
@@ -87,6 +87,34 @@
     "defaultMessage": "Violent Crimes",
     "description": "Label used to describe statistical variables in chart legends, string should be short. Describes the total number of violent crimes."
   },
+  "Count_CriminalIncidents_IsHateCrime": {
+    "defaultMessage": "Hate Crime Incidents",
```
Comment (beets) [inline static/js/i18n/strings/en/stats_var_labels.json:91]: "Hate Crimes" is more succinct and would work here too

File: static/js/i18n/strings/en/stats_var_labels.json [hunk 3]
```diff
@@ -87,6 +87,34 @@
...
+    "description": "Label used to describe statistical variables in chart legends, string should be short. Describes the total number of hate crime incidents."
+  },
+  "Count_CriminalIncidents_BiasMotivationGender_IsHateCrime": {
+    "defaultMessage": "Hate Crime Incidents",
```
Comment (beets) [inline static/js/i18n/strings/en/stats_var_labels.json:95]: A more specific label would be good - "Hate Crimes (Motivated by Gender)"

The following labels can follow the same template

File: static/js/i18n/strings/en/stats_var_titles.json [hunk 4]
```diff
@@ -87,6 +87,34 @@
     "defaultMessage": "Number of Criminal Activities (Violent Crimes)",
     "description": "Title for a page about a statistical variable. Variable describes total number of violent crimes in a place."
   },
+  "Count_CriminalIncidents_IsHateCrime": {
+    "defaultMessage": "Number of Criminal Hate Crime Incidents",
```
Comment (beets) [inline static/js/i18n/strings/en/stats_var_titles.json:91]: "Number of Criminal Incidents (Hate Crimes)"

File: static/js/i18n/strings/en/stats_var_titles.json [hunk 5]
```diff
@@ -87,6 +87,34 @@
...
+    "description": "Title for a page about a statistical variable. Variable describes total number of hate crime incidents in a place."
+  },
+  "Count_CriminalIncidents_BiasMotivationGender_IsHateCrime": {
+    "defaultMessage": "Number of Criminal Hate Crime Incidents",
```
Comment (beets) [inline static/js/i18n/strings/en/stats_var_titles.json:95]: Template for this and the following:
"Number of Criminal Incidents (Hate Crimes Motivated by Gender)"

**Review by beets — COMMENTED** [review]
A few updates to the format of new strings.

File: static/js/i18n/strings/en/stats_var_titles.json (hunk 5 above)
Comment (rbhoot) [reply static/js/i18n/strings/en/stats_var_titles.json]: fixed.
I totally missed changing the defaultMessage string. 

Thanks for the review and support @beets :)

File: static/js/i18n/strings/en/stats_var_titles.json (hunk 4 above)
Comment (rbhoot) [reply static/js/i18n/strings/en/stats_var_titles.json]: fixed.

File: static/js/i18n/strings/en/stats_var_labels.json (hunk 3 above)
Comment (rbhoot) [reply static/js/i18n/strings/en/stats_var_labels.json]: fixed.

File: server/config/chart_config/crime.json (hunk 1 above)
Comment (rbhoot) [reply server/config/chart_config/crime.json]: I had skipped those because the numbers might seem too large if population is small. Added now.

**Comment by rbhoot** [comment]
image
image
image
image
image
image

File: server/i18n/all.pot [hunk 6]
```diff
@@ -428,6 +428,62 @@ msgstr "Government expenditures on military (% of GDP)"
 msgid "CHART_TITLE-Graduate_or_professional_degree_rate_gender"
 msgstr "Rate of graduate or professional degree attainment by gender"
 
+#. Title of a place chart: Total number of hate crime incidents
+msgid "CHART_TITLE-Hate_Crime"
+msgstr "Hate Crime"
```
[the remark this answers is not in the document]
Comment (rbhoot) [reply server/i18n/all.pot]: fixed.

**Comment by rbhoot** [comment]
The tests seem to break because of an import error. `ImportError: cannot import name 'soft_unicode' from 'markupsafe'`

**Comment by beets** [comment]
sorry for the late review -- the markupsafe issue was fixed in #1456. it looks like there is a small failure in a json file to be fixed. the changes otherwise look good, thanks for persisting!

**Comment by rbhoot** [comment]
Thanks for the feedback, I had missed fixing capitalisation in the golden file. Fixed it now. Thanks for the help during the PR.

File: server/i18n/all.pot [hunk 7]
```diff
@@ -428,6 +428,62 @@ msgstr "Government expenditures on military (% of GDP)"
...
+
+#. Title of a place chart: Total number of hate crime incidents motivated by disability
+msgid "CHART_TITLE-Hate_Crime_Disability"
+msgstr "Hate Crime motivated by disability"
+
+#. Title of a place chart: Total number of hate crime incidents motivated by disability per 100,000 people
+msgid "CHART_TITLE-Hate_Crime_Disability_rate"
+msgstr "Hate crimes motivated by disability per 100,000 people"
```
[the remark this answers is not in the document]
Comment (beets) [reply server/i18n/all.pot:441]: Note, this needs to be done in the json.

**Review by beets — APPROVED** [review]
hi ravi - there are still some "hate crimes" rather than "hate crime" in the json, so it would be great if you can fix it (or we can do that as a follow up after this goes in).

**Comment by rbhoot** [comment]
Hi Carolyn, if you are pointing to the `defaultMessage` in the statvar title and label, i had kept those consistent with others on top `Number of Criminal Activities (Violent Crimes)`. I can change the ones you recommend.

**Comment by beets** [comment]
thanks for the updates @rbhoot!

**Pull request merged**